server/src/http/unleash_client.rs

```diff
@@ -541,6 +561,48 @@ impl UnleashClient {
         }
     }
 
+    #[tracing::instrument(skip(self, instance_data, token))]
+    pub async fn send_instance_data(
+        &self,
+        instance_data: EdgeInstanceData,
+        token: &str,
+    ) -> EdgeResult<()> {
+        let started_at = Utc::now();
+        let result = self
+            .backing_client
+            .post(self.urls.edge_instance_data_url.to_string())
+            .headers(self.header_map(Some(token.into())))
+            .timeout(Duration::seconds(3).to_std().unwrap())
+            .json(&instance_data)
+            .send()
+            .await
+            .map_err(|e| {
+                info!("Failed to send instance data: {e:?}");
+                EdgeError::EdgeMetricsError
+            })?;
+        let ended_at = Utc::now();
+        INSTANCE_DATA_UPLOAD
+            .with_label_values(&[result.status().as_str()])
+            .observe(
+                ended_at
+                    .signed_duration_since(started_at)
+                    .num_milliseconds() as f64,
+            );
+        let r = if result.status().is_success() {
+            Ok(())
+        } else {
+            match result.status() {
+                StatusCode::BAD_REQUEST => Err(EdgeMetricsRequestError(
```

Comment:
Why this specific check?

Reply:
Because copy-pasta from earlier request. I still like it though, because it exposes full message if we receive 400 from upstream, so we know what we need to fix.